cmake/kokkos_arch.cmake

```
@@ -484,6 +484,43 @@ IF(KOKKOS_ENABLE_CUDA AND NOT CUDA_ARCH_ALREADY_SPECIFIED)
     ${CMAKE_CURRENT_SOURCE_DIR}/cmake/compile_tests/cuda_compute_capability.cc
     COMPILE_DEFINITIONS -DSM_ONLY
     RUN_OUTPUT_VARIABLE _CUDA_COMPUTE_CAPABILITY)
+
+  # if user is using kokkos_compiler_launcher, above will fail.
+  IF(NOT _COMPILE_RESULT OR NOT _RESULT EQUAL 0 OR FLAG_INDEX EQUAL -1)
+    # check to see if CUDA is not already enabled (may happen when Kokkos is subproject)
+    GET_PROPERTY(ENABLED_LANGUAGES GLOBAL PROPERTY ENABLED_LANGUAGES)
```

Can we use a slightly different name for the variable to avoid confusions?
```suggestion
    GET_PROPERTY(_ENABLED_LANGUAGES GLOBAL PROPERTY ENABLED_LANGUAGES)
```

---

Well the property isn't a global variable. Like if you were to do `message(STATUS "ENABLED_LANGUAGES: ${ENABLED_LANGUAGES}")` before the `get_property` you would get: `-- ENABLED_LANGUAGES: `.

---

Yes, I know. I just find it harder to read naming the variable the same as the property you are asking for.